src/qz/ui/AboutDialog.java

```diff
@@ -1,150 +1,200 @@
+            BufferedReader br = new BufferedReader(new InputStreamReader(api.openStream()));
 
-        gridPanel.add(createLabel("Secure Socket:", true));
-        gridPanel.add(wssLabel);
+            StringBuilder rawJson = new StringBuilder();
+            String line;
+            while((line = br.readLine()) != null) {
+                rawJson.append(line);
+            }
 
-        gridPanel.add(createLabel("Fallback Socket:", true));
-        gridPanel.add(wsLabel);
+            JSONArray json = new JSONArray(rawJson.toString());
+            latestVersion = Version.valueOf(json.getJSONObject(0).getString("name"));
 
-        gridPanel.add(createLabel("Publisher:", true));
-        try {
-            gridPanel.add(new LinkLabel(new URL(Constants.ABOUT_URL)));
+            log.trace("Found latest version: {}", latestVersion);
         }
-        catch(MalformedURLException ex) {
-            gridPanel.add(new LinkLabel(Constants.ABOUT_URL));
+        catch(Exception e) {
+            log.error("Failed to get latest version info", e);
         }
-        refresh();
-        setContent(pane, true);
     }
 
-    @Override
-    public void refresh() {
-        wssLabel.setForeground(wssLabel.getText().startsWith("wss:") ? Constants.TRUSTED_COLOR : Constants.WARNING_COLOR);
-        shadeComponents();
-        super.refresh();
+    public void usePort(int port) {
+        connectionPort = port;
+
+        initComponents();
     }
 
-    public void shadeComponents() {
-        Color bg = UIManager.getColor("Panel.background");
-        for(int i = 0; i < gridPanel.getComponents().length; i++) {
-            if (i % 4 == 0 || i % 4 == 1) {
-                if (gridPanel.getComponent(i) instanceof JComponent) {
-                    ((JComponent)gridPanel.getComponent(i)).setOpaque(true);
-                    gridPanel.getComponent(i).setBackground(SystemUtilities.isDarkMode() ? bg.darker() : bg.brighter());
-                }
+    public void initComponents() {
+        setIconImage(getImage(IconCache.Icon.ABOUT_ICON));
+
+        //language=HTML
+        String styleBlock = "<style> " +
+                "  body { font-family: 'sans-serif'; font-size: 10px; } " +
+                "  h1 { font-size: 36px; font-weight: normal; } " +
+                "  div { margin: 5px 0; width: 100%; } " +
+                "</style>";
+
+        //language=HTML
+        String infoText = "<h1>%title%</h1>" +
+                "<div>%version% (Java) <a href='%version_url%'>What's New?</a></div>" +
+                "<div>%update_action%</div>" +
+                "#split#" + //Split text to accommodate swing buttons
+                "<div>%title% is written and supported by %company%.</div>" +
+                "<div>If using %title% commercially, please first reach out to the website publisher for support issues.</div>" +
+                "<div><a href='%library_url%'>Detailed library information \uD83D\uDDD7</a></div>";
```

Comment:
The unicode character doesn't render on my Mac for some reason.

<img width="209" alt="Screen Shot 2019-09-17 at 10 00 58 AM" src="https://user-images.githubusercontent.com/6345473/65049657-1f68cf00-d934-11e9-92cd-c6ebb5f20771.png">

Reply:
I've just removed this icon entirely since all of the links open in a web browser anyways.